js/background.js

```diff
@@ -1,80 +1,76 @@
+        }
     }
-  });
 }
 
-// This is to handle the status updation when the app is "loaded" or "refreshed". 
-checkSession(); 
+/**
+ * Get the host name from a given URL
+ * @param href - the URL taken from the address bar
+ * @returns {string} hostname - the host name extracted from the URL
+ */
+var getHostName = function (href) {
+    var l = document.createElement("a");
+    l.href = href;
+    return l.hostname;
+};
 
-/* Updates the status especially when the 'channeli.in' tabs are updated */
-chrome.tabs.onUpdated.addListener(function(tabId, changeInfo, tab) {
-  var tab_url = tab.url;
-  var domain = getDomainName(tab_url);
-  if (tab_url !== undefined && changeInfo.status == "complete") {
-    if(domain == Host) {
-      checkSession();
-      /*****************************************************************************************
-      //TODO: When more than one tab from 'channel.in' domain are active, then logging out
-      //      from channeli in one tab should automatically make other tabs from the same  
-      //      domain to refresh along with the extension.
-
-      chrome.tabs.query({}, function (tabs) {
-          var _tabs = [];
-          for (var i = 0; i < tabs.length; i++) {
-            if(getDomainName(tabs[i].url) == Host)
-            {
-              if(tabs[i].id == tabId) continue; 
-              else {
-                _tabs.push(tabs[i].id);
-              }
-            }
-          }
-          for(var j = 0; j < _tabs.length; j++)
-          {
-            
-            chrome.tabs.reload(_tabs[j]);
-          }
-      });
-      ******************************************************************************************/
-    }
-  }
-}); 
-
-var NetworkStatus = 0; /* 0 - offline, 1 - online */
+/** Check if the user is logged in by performing a request on the LecTut API */
+var checkSession = function () {
+    var oldUserStatus = userStatus;
+    var url = DOMAIN + "/lectut_api/";
+    $.get(url, function (data) {
+        var userType = data.userType;
+        if (userType == 0) {
+            // Logged in
+            userStatus = 1;
+            chrome.browserAction.setIcon({path: "../images/icon_active.png"});
+        } else {
+            // Not logged in
+            userStatus = 0;
+            chrome.browserAction.setIcon({path: "../images/icon_inactive.png"});
+        }
+    });
+};
 
-  /* Checks Network Connection Status */
-  var checkNetConnection = function() {
-    $.get(Domain, {}, function(res){
-      if(NetworkStatus == 0) {
-        checkSession();
-        NetworkStatus = 1;
-      }
-    })
-    .fail( function(res) {
-      NetworkStatus = 0;
-      chrome.browserAction.setIcon({path: "../images/icon_inactive.png"});
+/** Check if the user is connected to the Internet by contacting the CERN webpage */
+var checkNetConnection = function () {
+    $.get(NET_CHECK_DOMAIN, {}, function () {
+        if (networkStatus == 0) {
+            checkSession(-1);
```

Comment:
Why 'checkSession' is accepting an argument '-1' here ? Where is the corresponding function declaration for this call ?